Xamarin.PropertyEditing/Reflection/ReflectionObjectEditor.cs

```diff
@@ -138,6 +127,31 @@ public Task<IReadOnlyList<ITypeInfo>> GetAssignableTypesAsync (IPropertyInfo pro
 			};
 		}
 
+		internal static Task<AssignableTypesResult> GetAssignableTypes (IPropertyInfo property, bool childTypes)
+		{
+			return Task.Run (() => {
+				var types = AppDomain.CurrentDomain.GetAssemblies ().SelectMany (a => a.GetTypes ()).AsParallel ()
+					.Where (t => t.Namespace != null && !t.IsAbstract && !t.IsInterface && t.IsPublic && t.GetConstructor (Type.EmptyTypes) != null);
+
+				Type type = property.Type;
+				if (childTypes) {
+					var generic = property.Type.GetInterface ("ICollection`1");
+					if (generic != null) {
+						type = generic.GetGenericArguments()[0];
+					} else {
+						type = typeof(object);
+					}
+				}
+
+				types = types.Where (t => type.IsAssignableFrom (t));
+
+				return new AssignableTypesResult (types.Select (t => {
+					string asmName = t.Assembly.GetName ().Name;
+					return new TypeInfo (new AssemblyInfo (asmName, isRelevant: asmName.StartsWith ("Xamarin")), t.Namespace, t.Name);
```

Comment:
Should we have a comment in there about why only types in Xamarin assemblies are considered relevant?